sktime/forecasting/base/_base.py

```diff
@@ -170,6 +170,27 @@ def __rmul__(self, other):
         else:
             return NotImplemented
 
+    def __or__(self, other):
+        """Magic | method, return MultiplexForecaster.
+
+        Implemented for `other` being either a MultiplexForecaster or a forecaster.
+
+        Parameters
+        ----------
+        other: `sktime` forecaster or sktime MultiplexForecaster
+
+        Returns
+        -------
+        MultiplexForecaster object
+        """
+        from sktime.forecasting.compose import MultiplexForecaster
+
+        # generate a name for our forecaster based on class name:
+        self_name = type(self).__name__.lower()
+        # create wrap self in MultiplexForecaster, pass most of '|' logic to that class
+        multiplex_self = MultiplexForecaster([(self_name, self)])
+        return multiplex_self | other
```

Comment:
we need to make a case distinction - you do the above if `other` is an instance of `BaseForecaster`, otherwise you need to return `NotImplemented` (see the other dunders)

Reply:
So - two things I would appreciate some clarity on!  
1) I actually intententionally had the code return a `ValueError` if other isn't a `BaseForecaster` or a `MultiplexForecaster` instead of `NotImplemented`. (My impression is that [they plan to phase out NotImplemented](https://docs.python.org/3/library/constants.html#NotImplemented), so I decided to just use an error instead).  But given that would you prefer I stick with the `NotImplemented` all the same?   
2) is your issue that there are not checks (as there are - they are just wrapped up in the `MultiplexForecaster` `__or__`) or that you want them in 2 places?  I can understand the logic for the second as it makes it clearer what is going on at all stages, but does lead to code bloat a bit too..... Hence the clarification question :)

Reply:
Whoops!  Just kidding - I think I misread the python docs page!  Just went back and reread it!  I guess they are just depreciating the use of it in a boolean context not all together!

Reply:
In which case I am happy to stick with it since that is what the other dunders use!  Though personally I might find a `ValueError` more useful haha.

Reply:
I think the convention with dunders is returning `NotImplemented` if they are not implemented, and it is important to do so.

Why? There is a dunder parse order, let's say we have an expression `a * b * c`. If `a` does not implement `__mul__` for `b`, then python looks next whether `__rmul__` of `b` implements for `a`, which can be the case, and so on.

Had you raised an exception in `a.__mul__`, this would have broken in the first step of parsing. But, expected behaviour and implementation is that python needs to check whether one of the two objects implements it. The information is passed via `NotImplemented`, without raising an exception.

Reply:
No worries - it is implemented like that in the new code! :)

Reply:
But this is helpful to know/makes sense